### ISSUE TITLE
building with PHP 8.2 failed

### ISSUE THREAD

**Issue by andypost** [swoole/swoole-src#4781]
Please answer these questions before submitting your issue.

1. What did you do? If possible, provide a simple script for reproducing the error.
```
	case "$CARCH" in
		ppc64le|s390x) export LDFLAGS="$LDFLAGS -lucontext" ;;
	esac
	phpize82
	./configure --prefix=/usr \
		--with-php-config=php-config82 \
		--enable-mysqlnd \
		--enable-openssl --with-openssl-dir=/usr \
		--enable-sockets \
		--enable-swoole-curl \
		--enable-cares
	make
```

2. What did you expect to see?
the build is passed

3. What did you see instead?
```
/home/skilld/aports/testing/php82-pecl-swoole/src/swoole-5.0.0/thirdparty/php/curl/interface.cc: In function 'void swoole_native_curl_minit(int)':
/home/skilld/aports/testing/php82-pecl-swoole/src/swoole-5.0.0/thirdparty/php/curl/interface.cc:282:57: error: invalid conversion from 'int (*)(zend_object*, zval*, int)' {aka 'int (*)(_zend_object*, _zval_struct*, int)'} to 'zend_object_cast_t' {aka 'ZEND_RESULT_CODE (*)(_zend_object*, _zval_struct*, int)'} [-fpermissive]
  282 |     swoole_coroutine_curl_handle_handlers.cast_object = swoole_curl_cast_object;
      |                                                         ^~~~~~~~~~~~~~~~~~~~~~~
      |                                                         |
      |                                                         int (*)(zend_object*, zval*, int) {aka int (*)(_zend_object*, _zval_struct*, int)}
/home/skilld/aports/testing/php82-pecl-swoole/src/swoole-5.0.0/thirdparty/php/curl/multi.cc: In function 'void curl_multi_register_class(const zend_function_entry*)':
/home/skilld/aports/testing/php82-pecl-swoole/src/swoole-5.0.0/thirdparty/php/curl/multi.cc:601:63: error: invalid conversion from 'int (*)(zend_object*, zval*, int)' {aka 'int (*)(_zend_object*, _zval_struct*, int)'} to 'zend_object_cast_t' {aka 'ZEND_RESULT_CODE (*)(_zend_object*, _zval_struct*, int)'} [-fpermissive]
  601 |     swoole_coroutine_curl_multi_handle_handlers.cast_object = swoole_curl_cast_object;
      |                                                               ^~~~~~~~~~~~~~~~~~~~~~~
      |                                                               |
      |                                                               int (*)(zend_object*, zval*, int) {aka int (*)(_zend_object*, _zval_struct*, int)}
make: *** [Makefile:531: thirdparty/php/curl/interface.lo] Error 1
make: *** Waiting for unfinished jobs....
make: *** [Makefile:534: thirdparty/php/curl/multi.lo] Error 1
>>> ERROR: php82-pecl-swoole: build failed
```

4. What version of Swoole are you using (show your `php --ri swoole`)?
5.0.0 and PHP 8.2 beta1

5. What is your machine environment used (show your `uname -a` & `php -v` & `gcc -v`) ?
- `curl-dev (7.84.0-r1)`
- `5.15.0-43-generic #46-Ubuntu SMP Tue Jul 12 10:30:17 UTC 2022 x86_64 Linux`
```
~/aports/testing/php82-pecl-swoole $ php82 -v
PHP 8.2.0beta1 (cli) (built: Jul 28 2022 13:30:09) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.2.0-dev, Copyright (c) Zend Technologies
~/aports/testing/php82-pecl-swoole $ gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-alpine-linux-musl/11.2.1/lto-wrapper
Target: x86_64-alpine-linux-musl
Configured with: /home/buildozer/aports/main/gcc/src/gcc-11.2.1_git20220219/configure --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --build=x86_64-alpine-linux-musl --host=x86_64-alpine-linux-musl --target=x86_64-alpine-linux-musl --with-pkgversion='Alpine 11.2.1_git20220219' --enable-checking=release --disable-fixed-point --disable-libstdcxx-pch --disable-multilib --disable-nls --disable-werror --disable-symvers --enable-__cxa_atexit --enable-default-pie --enable-default-ssp --enable-cloog-backend --enable-languages=c,c++,d,objc,go,fortran,ada,jit --disable-libssp --disable-libmpx --disable-libmudflap --disable-libsanitizer --enable-shared --enable-threads --enable-tls --enable-host-shared --with-system-zlib --with-linker-hash-style=gnu
Thread model: posix
Supported LTO compression algorithms: zlib
gcc version 11.2.1 20220219 (Alpine 11.2.1_git20220219) 
```

**Comment by andypost** [comment]
Ref https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/37150

**Comment by matyhtf** [comment]
Fixed https://github.com/swoole/swoole-src/commit/e085097fb9001a3e8e78a47ca4541da2f675661c

**Comment by andypost** [comment]
@matyhtf thank you a lot! it builds fine, pushed for testing

**Comment by waghanza** [comment]
Seeking around for a solution to have `swoole` on **8.2**, I see that this fix is already merged and in `5.0.2` but this version is not yet in `pecl`

will it be soon ?
